docs/guides/managed-backup/dedicated-backup-namespace-auto-backup/index.md

```diff
@@ -0,0 +1,407 @@
+
+**Create Storage Secret:**
+
+At first, let's create a Storage Secret for the GCS backend in the `backup` namespace,
+
+```bash
+$ echo -n 'changeit' > RESTIC_PASSWORD
+$ echo -n '<your-project-id>' > GOOGLE_PROJECT_ID
+$ mv downloaded-sa-json.key GOOGLE_SERVICE_ACCOUNT_JSON_KEY
+$ kubectl create secret generic -n backup gcs-secret \
+    --from-file=./RESTIC_PASSWORD \
+    --from-file=./GOOGLE_PROJECT_ID \
+    --from-file=./GOOGLE_SERVICE_ACCOUNT_JSON_KEY
+secret/gcs-secret created
+```
+
+### Prepare RBAC
+
+Stash does not grant necessary RBAC permissions to the backup job for taking backup from a different namespace. In this case, we have to provide the RBAC permissions manually. This helps to prevent unauthorized namespaces from getting access to a database via Stash.
+
+**Create ServiceAccount:**
+
+At first, we are going to create a ServiceAccount in the `backup` namespace. This ServiceAccount will refer to the granted permissions for taking backup. Here is the YAML of the ServiceAccount,
+
+```yaml
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: cross-namespace-target-reader
+  namespace: backup
+```
+
+Let's create the ServiceAccount,
+
+```bash
+$ kubectl apply -f https://github.com/stashed/docs/raw/{{< param "info.version" >}}/docs/guides/managed-backup/dedicated-backup-namespace-auto-backup/examples/serviceaccount.yaml
+serviceaccount/cross-namespace-target-reader created
+```
+
+**Create ClusterRole and ClusterRoleBinding**
+
+We are going to create a ClusterRole and ClusterRoleBinding with the necessary permissions to perform the backup. Here are the YAMLs of the ClusterRole and ClusterRoleBinding,
+
+```yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: cross-namespace-target-clusterrole
+rules:
+- apiGroups: [""]
+  resources: ["secrets"]
+  verbs: ["get","list"]
+- apiGroups: ["appcatalog.appscode.com"]
+  resources: ["appbindings"]
+  verbs: ["get","list"]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: cross-namespace-target-clusterrolebinding
+subjects:
+- kind: ServiceAccount
+  name: cross-namespace-target-reader
+  namespace: backup
+roleRef:
+  kind: ClusterRole
+  name: cross-namespace-target-clusterrole
+  apiGroup: rbac.authorization.k8s.io
+```
+
+Let's create the ClusterRole and ClusterRoleBinding we have shown above,
+
+```bash
+$ kubectl apply -f https://github.com/stashed/docs/raw/{{< param "info.version" >}}/docs/guides/managed-backup/dedicated-backup-namespace-auto-backup/examples/clusterrole_clusterrolebinding.yaml
+clusterrole.rbac.authorization.k8s.io/cross-namespace-target-clusterrole created
+clusterrolebinding.rbac.authorization.k8s.io/cross-namespace-target-clusterrolebinding created
+```
+
+The above RBAC permissions will allow ServiceAccounts to grant necessary backup job permissions from any namespace.
```

Comment:
The above RBAC permissions will allow the ServiceAccounts to perform backup targets from any namespace.